README.rst

```diff
@@ -69,6 +69,168 @@ The *apidoc* extension uses the following configuration values:
+
+   apidoc_module_dir = '../../openstack'
+   apidoc_excluded_paths = [
+     'volume',
+     'tests'
+   ]
+   apidoc_output_dir = 'contributor/api'
+
+.. note::
+
+    You could also remove the ``[build_sphinx]`` section from ``setup.cfg`` if
+    you wished to build documentation with ``sphinx-build`` directly instead.
+
+There are a couple of changes here:
+
+- Replace ``autodoc_tree_index_modules`` with ``apidoc_module_dir``
+
+  ``autodoc_tree_index_modules`` is boolean option indicating whether to enable
+  generation; however, when enabled, docs are always generated for any module
+  in the top level directory (where ``setup.cfg`` is located). By comparison
+  the extension is already enabled by adding it the ``extension`` attribute of
+  ``conf.py``; however, it is necessary to state the directory that you wish to
+  document. As a result, it is necessary to remove
+  ``autodoc_tree_index_modules`` from ``setup.cfg`` and define
+  ``apidoc_module_dir`` in ``conf.py``.
+
+- Replace ``autodoc_tree_excludes`` with ``apidoc_excluded_paths``
+
+  Like ``apidoc_excluded_paths``, ``autodoc_tree_excludes`` is a list of
+  fnmatch-style paths describing files and directories to exclude relative to
+  the source directory. This means it's a like-for-like replacement so you can
+  simply move the option to ``conf.py``, update the paths (assuming
+  ``apidoc_module_dir`` is configured to anything except ``.``), and rename it.
+
+- Replace ``api_doc_dir`` with ``apidoc_output_dir``
+
+  As above, ``api_doc_dir`` functions exactly as ``apidoc_output_dir`` does.
+  Simply move the option to ``conf.py`` and rename it.
+
+Once done, your output should work exactly as before.
+
+autodoc
+~~~~~~~
+
+*autodoc* is not based on *sphinx-apidoc*. Fortunately it is possible to
+generate something very similar (although not identical!). Let's take
+`oslo.privsep`_ as an example, once again looking at minimal versions of
+``setup.cfg`` and ``doc/source/conf.py``:
+
+.. code-block:: ini
+
+   [build_sphinx]
+   all_files = 1
+   build-dir = doc/build
+   source-dir = doc/source
+
+   [pbr]
+   autodoc_index_modules = True
+   api_doc_dir = reference/api
+   autodoc_exclude_modules =
+     oslo_privsep.tests.*
+     oslo_privsep._*
+
+.. code-block:: python
+
+   extensions = ['']
+
+Once migrated, this would look like so:
+
+.. code-block:: ini
+
+   [build_sphinx]
+   all_files = 1
+   build-dir = doc/build
+   source-dir = doc/source
+
+.. code-block:: python
+
+   extensions = ['sphinxcontrib.apidoc']
+
+   apidoc_module_dir = '../../oslo_privsep'
+   apidoc_excluded_paths = ['tests', '_*']
+   apidoc_output_dir = 'reference/api'
+   apidoc_separate_modules = True
+
+Most of the changes necessary are the same as `autodoc_tree`_, with some
+exceptions.
+
+- Replace ``autodoc_index_modules`` with ``apidoc_module_dir``
+
+  Same as the step required for *autodoc_tree*, but you're removing
+  ``autodoc_index_modules`` instead of ``autodoc_tree_index_modules``.
```

Comment:
There's something to be said for just duplicating the instructions rather than asking people to read both sections.

Reply:
+1 Will do

Reply:
Done